src/components/screen/util/grid-generation-and-select.jsx

```diff
@@ -74,93 +75,165 @@ function GridGenerationAndSelect({
     }
 
     // Removes duplicates.
-    const localSelectedPlaylists = uniqWith(
-      [...localTarget, ...selectedPlaylistsCopy],
-      (firstPlaylist, secondPlaylist) =>
-        firstPlaylist["@id"] === secondPlaylist["@id"] &&
-        firstPlaylist.region === secondPlaylist.region
+    const localSelectedPlaylists = [
+      ...localTarget,
+      ...selectedPlaylistsCopy,
+    ].filter(
+      (playlist, index, self) =>
+        index ===
+        self.findIndex(
+          (secondPlaylist) =>
+            secondPlaylist["@id"] === playlist["@id"] &&
+            secondPlaylist.region === playlist.region
+        )
     );
-    setSelectedPlaylists(localSelectedPlaylists);
-    handleInput({ target: { value: localSelectedPlaylists, id: "playlists" } });
+
+    return localSelectedPlaylists;
+  }
+
+  useEffect(() => {
+    if (regions.length > 0) {
+      const promises = [];
+      regions.forEach(({ "@id": id }) => {
+        promises.push(
+          dispatch(
+            api.endpoints.getV2ScreensByIdRegionsAndRegionIdPlaylists.initiate({
+              id: screenId,
+              regionId: idFromUrl(id),
+              page: 1,
```

Comment:
You have to get all the pages... otherwise playlists above # 10 will be removed if the screen is opened and saved later on.

Reply:
oh good catch, I just wrote it and forgot 🙃